consensus/spos/bls/subroundEndRound_test.go

```diff
@@ -610,6 +629,152 @@ func TestSubroundEndRound_CheckSignaturesValidityShouldRetunNil(t *testing.T) {
+	t.Parallel()
+
+	sr := *initSubroundEndRound()
+
+	// set previous as finished
+	sr.SetStatus(2, spos.SsFinished)
+
+	// set current as finished
+	sr.SetStatus(3, spos.SsFinished)
+
+	res := sr.DoEndRoundJobByParticipant()
+	assert.False(t, res)
+}
+
+func TestSubroundEndRound_DoEndRoundJobByParticipant_ConsensusHeaderNotReceivedShouldReturnFalse(t *testing.T) {
+	t.Parallel()
+
+	sr := *initSubroundEndRound()
+
+	// set previous as finished
+	sr.SetStatus(2, spos.SsFinished)
+
+	// set current as not finished
+	sr.SetStatus(3, spos.SsNotFinished)
+
+	res := sr.DoEndRoundJobByParticipant()
+	assert.False(t, res)
+}
+
+func TestSubroundEndRound_DoEndRoundJobByParticipant_ShouldReturnTrue(t *testing.T) {
+	t.Parallel()
+
+	hdr := &block.Header{Nonce: 37}
+	sr := *initSubroundEndRound()
+	sr.Header = hdr
+	sr.AddReceivedHeader(hdr)
+
+	// set previous as finished
+	sr.SetStatus(2, spos.SsFinished)
+
+	// set current as not finished
+	sr.SetStatus(3, spos.SsNotFinished)
+
+	res := sr.DoEndRoundJobByParticipant()
+	assert.True(t, res)
+}
+
+func TestSubroundEndRound_IsConsensusHeaderReceived_NoReceivedHeadersShouldReturnFalse(t *testing.T) {
+	t.Parallel()
+
+	hdr := &block.Header{Nonce: 37}
+	sr := *initSubroundEndRound()
+	sr.Header = hdr
+
+	res, retHdr := sr.IsConsensusHeaderReceived()
+	assert.False(t, res)
+	assert.Nil(t, retHdr)
+}
+
+func TestSubroundEndRound_IsConsensusHeaderReceived_HeaderNotReceivedShouldReturnFalse(t *testing.T) {
+	t.Parallel()
+
+	hdr := &block.Header{Nonce: 37}
+	hdrToSearchFor := &block.Header{Nonce: 38}
+	sr := *initSubroundEndRound()
+	sr.AddReceivedHeader(hdr)
+	sr.Header = hdrToSearchFor
+
+	res, retHdr := sr.IsConsensusHeaderReceived()
+	assert.False(t, res)
+	assert.Nil(t, retHdr)
+}
+
+func TestSubroundEndRound_IsConsensusHeaderReceivedShouldReturnTrue(t *testing.T) {
+	t.Parallel()
+
+	hdr := &block.Header{Nonce: 37}
+	sr := *initSubroundEndRound()
+	sr.Header = hdr
+	sr.AddReceivedHeader(hdr)
+
+	res, retHdr := sr.IsConsensusHeaderReceived()
+	assert.True(t, res)
+	assert.Equal(t, hdr, retHdr)
+}
+
+func TestSubroundEndRound_IsOutOfTimeShouldReturnFalse(t *testing.T) {
+	t.Parallel()
+
+	sr := *initSubroundEndRound()
+
+	res := sr.IsOutOfTime()
+	assert.False(t, res)
+}
+
+func TestSubroundEndRound_IsOutOfTimeShouldReturnTrue(t *testing.T) {
+	t.Parallel()
+
+	// update rounder's mock so it will calculate for real the duration
+	container := mock.InitConsensusCore()
+	rounder := mock.RounderMock{RemainingTimeCalled: func(startTime time.Time, maxTime time.Duration) time.Duration {
+		currentTime := time.Now()
+		elapsedTime := currentTime.Sub(startTime)
+		remainingTime := maxTime - elapsedTime
+
+		return remainingTime
+	}}
+	container.SetRounder(&rounder)
+	sr := *initSubroundEndRoundWithContainer(container)
+
+	sr.RoundTimeStamp = time.Now().AddDate(0, 0, -1)
+
+	res := sr.IsOutOfTime()
+	assert.True(t, res)
+}
+
```

Comment:
Can you check after merge with development, the code coverage again, as there are 2-3 more methods which have entered with the last consensus improvement, which also could need some unit tests. 

receivedBlockHeaderFinalInfo
haveConsensusHeaderWithFullInfo
createAndBroadcastHeaderFinalInfo

Reply:
done